jscomp/frontend/ast_utf8_string.ml

```diff
@@ -166,7 +176,24 @@ and unicode loc buf s offset s_len =
 (* http://www.2ality.com/2015/01/es6-strings.html
    console.log('\uD83D\uDE80'); (* ES6*)
    console.log('\u{1F680}');
-*)   
+*)  
+
+(* ES6 unicode codepoint escape sequences: \u{…} 
+   https://262.ecma-international.org/6.0/#sec-literals-string-literals *)
+and unicode_codepoint_escape loc buf s offset s_len =
+  if offset >= s_len then error ~loc Invalid_unicode_codepoint_escape
+  else
+    let cur_char = s.[offset] in
+    match cur_char with
+    | '}' ->
+      Buffer.add_char buf cur_char ;
+      check_and_transform (loc + 1) buf s (offset + 1) s_len
+    | _ ->
+      if Ext_char.valid_hex cur_char then begin
+        Buffer.add_char buf cur_char ;
+        unicode_codepoint_escape (loc + 1) buf s (offset + 1) s_len
+      end else
+        error ~loc Invalid_unicode_codepoint_escape
```

Comment:
do we have length limitation for things inside `\u{...}`?

Reply:
`"\u{abcabcabc}"`
seems to be a syntax error in JS, it is caught here?

Reply:
There is no length limitation for `\u{...}`. Technically user land could write: `\u{0000000000000000000000000007a}`.

Will throw a syntax error when the codepoint is invalid.

Reply:
this may still be wrong, since too long unicode chars will result in an overflow, we may need check that possibility

Reply:
@bobzhang Do you have a suggestion on how to handle the overflow case? Is there something available in the compiler repo? Or should I roll my own thing?

Reply:
You can check `x` with a conesrvative upperbound, e.g `LARGEST_UNICODE_POINT/16`